framework_tvb/tvb/core/neocom/_h5loader.py

```diff
@@ -105,9 +105,12 @@ def find_file_name(self, gid):
         fname = self._locate(gid)
         return fname
 
-    def load(self, gid):
+    def load(self, gid=None, fname=None):
         # type: (typing.Union[uuid.UUID, str]) -> HasTraits
-        fname = self.find_file_name(gid)
+        if fname is None:
+            if gid is None:
+                raise ValueError("Neither gid nor filename is provided to load!")
+            fname = self.find_file_name(gid)
```

Comment:
I suggest here to use the same paradign as in load, because we are in DirLoader, and we already have self.base_dir
`path = os.path.join(self.base_dir, fname)`

Reply:
I don't get it because both input fname and self.find_file_name(gid) return a file name, not a path
therefore it is enough to have join in here:
 with H5File.from_file(os.path.join(self.base_dir, fname)) as f:
Instead, in store() self.path_for(h5file_cls, datatype.gid) returns a full path, not just a file name